assets/js/modules/analytics-4/datastore/custom-dimensions.test.js

```diff
@@ -276,6 +276,92 @@ describe( 'modules/analytics-4 custom-dimensions', () => {
 				} );
 			} );
 		} );
+
+		describe( 'scheduleSyncAvailableCustomDimensions', () => {
+			it( 'schedules a sync request to run after two seconds', async () => {
+				registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
+					propertyID,
+				} );
+
+				fetchMock.postOnce(
+					new RegExp(
+						'^/google-site-kit/v1/modules/analytics-4/data/sync-custom-dimensions'
+					),
+					{
+						body: customDimensionNames,
+						status: 200,
+					}
+				);
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.scheduleSyncAvailableCustomDimensions();
+
+				const WAIT_AFTER_DISPATCH = 2500;
+
+				// Wait for more than two seconds as the sync is run after two seconds.
+				await new Promise( ( resolve ) => {
+					setTimeout( resolve, WAIT_AFTER_DISPATCH );
+				} );
```

Comment:
We should use jest fake timers and related functions to advance the time or run all timers rather than doing a real wait for this long in the tests.

Reply:
Updated, thank you!